### PR TITLE
master requires Python 2.6: clean up tasks

### PR DESCRIPTION
**Pull request by sa2ajj**
Since master requires Python 2.6, certain things could be cleaned up:

* `with` statement is available
* `json` module is part of Python and it's fast
* certain functions `string` module offers could be used directly on string instances
* `io` module is available (using `BytesIO` since `StringIO` is for unicode strings only; this needs to be reviewed in future)

In addition, imports are re-ordered according to PEP-8 (with exceptions :smile:)

### CODE REVIEW

**Comment by tardyp** [comment]
damn, you did that manually?

I think we should rather automate it in fiximports.py

It is possible that @stibbons has made some changes to fiximports for doing that.

It should be quite easy.

**Comment by Stibbons** [comment]
I would love to. +2

**Comment by sa2ajj** [comment]
automatic way would be nice...

I decided to look into a few other legacy things and will push them a bit later (hence the PR subject change).

**Comment by Stibbons** [comment]
Oh, for information, I have packaged fiximports for sublime users: https://packagecontrol.io/packages/Python%20Fix%20Imports

**Comment by sa2ajj** [comment]
I wish I used sublime text :smile:

**Comment by Stibbons** [comment]
And do you plan dropping python 2.6 support for bb9? It's been 8 years since latest version...

**Comment by sa2ajj** [comment]
At the moment, it's all about Twisted :/

**Comment by tardyp** [comment]
I'm worried that this  takes a bit too long, and require lots of rebase
I see that the HEAD~ of this branch succeed running the unit tests, I would be inclined to merge this and let you finish the rest of the wip.

**Comment by sa2ajj** [comment]
@tardyp oops, I thought I'd have finished it earlier.

Let me review the last change (it was not clear for me whether it's necessary at all) and, hopefully, we'll have this branch ready today.

**Comment by sa2ajj** [comment]
rebase was trivial.

Let's wait for the test results.

**Comment by sa2ajj** [comment]
@tardyp I think it's good to go.  (I'll look at other things in a new PR.)

**Comment by sa2ajj** [comment]
One more rebase after integrating #2057

**Comment by tardyp** [comment]
It still has rebase issues :-(

I will merge the import reordering alone, and let you rebase, so that we can review the rest in gh. right now, gh will refuse to review everything.

**Comment by tardyp** [comment]
https://github.com/buildbot/buildbot/pull/2077 has been merged. please rebase without your initial commit.

**Comment by Stibbons** [comment]
Was building the pypi package for fiximports: https://pypi.python.org/pypi/fiximports
Works great, but I discovered a much more flexible package: https://pypi.python.org/pypi/isort
I could even more split the packages into groups... and add a nice comment on imports groups

**Comment by tardyp** [comment]
@sa2ajj can you rebase this branch and re-upload? so that we can discuss on the diff

**Comment by sa2ajj** [comment]
@Stibbons that's a nice tool (`isort`)

@tardyp I'm going to do this before doing anything else.  Sorry.

**Comment by sa2ajj** [comment]
done

**Comment by sa2ajj** [comment]
now fixed the missed upon rebase use of `buildbot.util.json`

File: master/buildbot/config.py [hunk 1]
```diff
@@ -12,24 +12,23 @@
 # Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 # Copyright Buildbot Team Members
-from future.utils import iteritems
-from future.utils import itervalues
+from future.utils import (iteritems,
+                          itervalues)
```
Comment (tardyp) [inline master/buildbot/config.py]: I think that fiximport will destroy this change. did you run it?

File: master/buildbot/db/model.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 # Copyright Buildbot Team Members
-
+from migrate import exceptions
```
Comment (tardyp) [inline master/buildbot/db/model.py]: fiximport will split the "from"s from the "import"s

**Comment by tardyp** [comment]
I think you should install the git hook so that fiximport runs.
There is no point doing import fixes if they are destroyed if somebody touch the file and run fiximport..

'make hook' should do the trick.

**Comment by sa2ajj** [comment]
@tardyp thanks!  I keep forgetting about that :(

**Comment by tardyp** [comment]
looks like there is still a few pep8 issues and test failed

File: master/buildbot/test/integration/test_worker_transition.py [hunk 3]
```diff
@@ -21,15 +21,13 @@
...
 
-import buildbot.worker
-
 from buildbot import config
 from buildbot.master import BuildMaster
 from buildbot.test.util import dirs
 from buildbot.test.util import www
 from buildbot.test.util.warnings import assertNotProducesWarnings
 from buildbot.test.util.warnings import assertProducesWarning
-from buildbot.test.util.warnings import assertProducesWarnings
```
Comment (ewongbb) [inline master/buildbot/test/integration/test_worker_transition.py]: Was this supposed to be removed?

File: master/buildbot/util/__init__.py [hunk 4]
```diff
@@ -172,26 +172,6 @@ def ascii2unicode(x):
...
-    _tmp = json.loads
-except AttributeError:
-    import warnings
-    import sys
-    warnings.warn("Use simplejson, not the old json module.")
-    sys.modules.pop('json')  # get rid of the bad json module
-    import simplejson as json
-
 
 def toJson(obj):
```
Comment (ewongbb) [inline master/buildbot/util/__init__.py]: I realize this was already here before this, but I only just noticed this part of the code and I'm confused with the name of this method.  Why is it toJson when it converts a datetime to epoch, so
shouldn't it be something like toEpoch()?  Can someone clarify?

**Comment by tardyp** [comment]
ping.

**Comment by djmitche** [comment]
closing based on http://trac.buildbot.net/wiki/StalePullRequestPolicy -- a shame, as this was a nice cleanup, but it's quite bitrotten by now :(

**Pull request closed without merging**